### ISSUE TITLE
axe reports a11y issues on calcite-slider

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-components#1156]
### Summary

The following violations are reported by axe on `calcite-slider`:

1. [ ] ARIA role should be appropriate for the element (for both single and double thumbs)
2. [ ] Ensures buttons have discernible text

#### Details

```
ARIA role should be appropriate for the element

Ensures role attribute has an appropriate value for the element

[More info...](https://dequeuniversity.com/rules/axe/4.3/aria-allowed-role?application=axeAPI)
#calcite-slider-3343e91d-32b7-2975-96f1-06dfc91cb47d.thumb

Minor
ARIA role slider is not allowed for given element
cat.aria
best-practice
```

```
Ensures buttons have discernible text

[More info...](https://dequeuniversity.com/rules/axe/4.3/button-name?application=axeAPI)
#calcite-slider-3343e91d-32b7-2975-96f1-06dfc91cb47d.thumb

Critical
Element does not have inner text that is visible to screen readers
Serious
aria-label attribute does not exist or is empty
Serious
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
Serious
Element has no title attribute
Minor
Element's default semantics were not overridden with role="none" or role="presentation"
```

![Screen Shot 2021-12-28 at 11 11 12 AM](https://user-images.githubusercontent.com/197440/147599494-a889cfef-92ca-4a15-97b3-0bd0d606e56c.png)

### Actual Behavior

The above issues are reported by axe.

### Expected Behavior

No more axe violations reported on the slider.

### Reproduction Steps

1. Go to [storybook](https://esri.github.io/calcite-components/?path=/story/components-controls-slider--single-value)
1. Go to the A11y tab.
1. Click on the "Tests completed" button in the bottom right to re-run tests.
1. Toggle a knob.
1. Repeat clicking on "Tests completed"
1. Do for each story and knob

**Comment by paulcpederson** [comment]
@driskull thanks for writing these up, we'll probably need an additional prop like `intl-year` on the date component so we can get a localized string for this. The card will probably need props for strings as well. I can see app devs needing something like "select item {title}" and "deselect item {title}".

**Comment by driskull** [comment]
These could also be tested by...

- using e2e tests with the `accessible` helper.
- using the aXe toolbar to run on storybook or demo pages.

Ideally, we have the `accessible` e2e helper for all components but I don't think we could get 100% coverage because of all the combinations of props & attributes but if we did the majority or high impact ones we would be good.

**Comment by jcfranco** [comment]
> Ideally, we have the accessible e2e helper for all components but I don't think we could get 100% coverage because of all the combinations of props & attributes but if we did the majority or high impact ones we would be good.

Agreed. https://github.com/Esri/calcite-components/issues/964 could also lessen the burden on `accessible` tests.

**Comment by jcfranco** [comment]
@driskull Can you confirm if the latest version of the components still have issues?

**Comment by driskull** [comment]
@jcfranco it seems like slider is the only one still having an aXe issue

**Comment by jcfranco** [comment]
@driskull Awesome, thanks for the update. I'll repurpose this for the remaining issue.

**Comment by benelan** [comment]
verified on beta.79 - all a11y tests are passing on storybook using the repro steps provided
